### PR TITLE
Fix #1694 api2.branch.io

### LINKED ISSUE #1694: api2.branch.io
### Issue URL (Incorrect Blocking)
 [http://branchi.io](https://adguardteam.github.io/AnonymousRedirect/redirect.html?url=http%3A%2F%2Fbranchi.io)

### Screenshots
<details>
  <summary>Screenshot 1 </summary>

  Screenshot 1
</details>
<details>
  <summary>Screenshot 1 with AdGuard disabled</summary>

  Screenshot 1
</details>

### System configuration

Information | Value
--- | ---
AdGuard product: | AdGuard for iOS v4.5.3
System version: | ios 17.5.1
License type: | paid
Advanced protection: | enabled
DNS filtering: | server: `yours`, `adguard`<br>filters: `https://filters.adtidy.org/dns/filter_1_ios.txt`, `look all I'm trying to do is tell you that there's a syntax error and it messes up trying to allow branch.io`, `No matter how I add branch.io to the allow list there is a syntax error in the blacklist that says this`, `||api2.branch.io^$dnsrewrite=0.0.0.0`, `But it looks like there's a carriage return in between the two pipes and api2`, `No matter how I add it it's still blocks and does the DNS rewrite of 0.0.0.0`, ``
Filters: | <b>Ad Blocking:</b><br/>AdGuard Base, <br/>AdGuard Mobile Ads<br/><br/><b>Privacy:</b><br/>AdGuard Tracking Protection<br/><br/><b>Annoyances:</b><br/>AdGuard Cookie Notices, <br/>AdGuard Popups, <br/>AdGuard Mobile App Banners, <br/>AdGuard Other Annoyances, <br/>AdGuard Widgets, <br/>Adblock Warning Removal List<br/><br/><b>Other:</b><br/>AdGuard DNS, <br/>AdGuard Experimental, <br/>Filter unblocking search ads and self-promotion
Other extensions used: | none

### PR DESCRIPTION
**Pull request by jellizaveta**
https://github.com/AdguardTeam/AdGuardSDNSFilter/issues/1694

### CODE REVIEW

**Review by piquark6046 — APPROVED** [review]

**Comment by AdamWr** [comment]
I'm not sure about this - https://github.com/AdguardTeam/AdGuardSDNSFilter/issues/943#issuecomment-1147197456 and:
https://github.com/AdguardTeam/AdGuardSDNSFilter/blob/1b02fe19a89b48fa9b9d2c7eb47a01171cab54db/Filters/rules.txt#L576-L578

**Comment by Alex-302** [comment]
@AdamWr reddit app does not have such problems when this domain is blocked. It is installed on my device.

File: Filters/exclusions.txt [hunk 1]
```diff
@@ -253,7 +253,7 @@
 ! https://github.com/AdguardTeam/AdGuardSDNSFilter/issues/946
 ||customer.io^
 ! https://github.com/AdguardTeam/AdGuardSDNSFilter/issues/943
-||branch.io^
+branch.io
```
Comment (Alex-302) [inline Filters/exclusions.txt:256]: Not a goog idea. It is popular, but only few apps are broken.

**Comment by AdamWr** [comment]
> reddit app does not have such problems when this domain is blocked. It is installed on my device.

I just meant, that I wasn't sure if it should be unblocked, and if it would, then this rule probably should be removed as well.
And as far as I understand, the main problem from the report - https://github.com/AdguardTeam/AdGuardSDNSFilter/issues/1694 is:
>look all I'm trying to do is tell you that there's a syntax error and it messes up trying to allow branch.io, No matter how I add branch.io to the allow list there is a syntax error in the blacklist that says this

So user want to unblock it in user rules only, I don't see anything in the report why it should be unblocked in filters.

**Comment by jellizaveta** [comment]
> I'm not sure about this - [#943 (comment)](https://github.com/AdguardTeam/AdGuardSDNSFilter/issues/943#issuecomment-1147197456) and:
> 
> https://github.com/AdguardTeam/AdGuardSDNSFilter/blob/1b02fe19a89b48fa9b9d2c7eb47a01171cab54db/Filters/rules.txt#L576-L578

AdGuard popup filter is built by script, if run `yarn build` with updated exclusion this rule  `||api2.branch.io^$dnsrewrite=0.0.0.0` will not be in filter.

**Comment by AdamWr** [comment]
>AdGuard popup filter is built by script, if run yarn build with updated exclusion this rule ||api2.branch.io^$dnsrewrite=0.0.0.0 will not be in filter.

So it wouldn't be any reason to keep this rule and in such situation it rather should be removed.
Anyway, I think that nothing should be changed, for now at least.

File: Filters/exclusions.txt (hunk 1 above)
Comment (AdamWr) [reply Filters/exclusions.txt:256]: This rule unblocks `branch.io` main domain, which I think it's fine, so in my opinion we can leave it as it is.

Comment (jellizaveta) [reply Filters/exclusions.txt:256]: okay

**Pull request closed without merging**